main.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const s3 = require('./lib/s3.js');
+const image = require('./lib/image.js');
+const awsAuth = require('./lib/commonAWS');
+const async = require('async');
+const path = require('path');
+const fs = require('fs');
+
+const execute = (fileName, argv, callback) => {
+  // establish AWS credentials:
+  const aws = awsAuth(argv);
+  // make sure working directory is present:
+  if (!fs.existsSync) {
+    fs.mkdirSync('build');
```

Comment:
Imagemin wants to read/write things from file, so I had it make a temp working directory.